### PR TITLE
Add custom `cancel`

### PR DESCRIPTION
**Pull request by garyghayrat**
Implemented the cancel functionality, i.e. the proposer can cancel while the propsal is still in the Pending state.

### CODE REVIEW

File: test/L2ArbitrumGovernorV2.t.sol [hunk 1]
```diff
@@ -196,6 +206,96 @@ abstract contract Quorum is L2ArbitrumGovernorV2Test {
...
+    address[] targets,
+    uint256[] values,
+    string[] signatures,
+    bytes[] calldatas,
+    uint256 voteStart,
+    uint256 voteEnd,
+    string description
+  );
+
+  function testFuzz_Propose(uint256 _actorSeed) public {
```
Comment (wildmolasses) [inline test/L2ArbitrumGovernorV2.t.sol]: needs a different name

```suggestion
  function testFuzz_CreatesProposalAndEmitsEvent(uint256 _actorSeed) public {
```

File: test/L2ArbitrumGovernorV2.t.sol [hunk 2]
```diff
@@ -196,6 +206,96 @@ abstract contract Quorum is L2ArbitrumGovernorV2Test {
...
+    uint256 proposalId = governor.hashProposal(targets, values, calldatas, keccak256(bytes(description)));
+    address _actor = _getMajorDelegate(_actorSeed);
+    vm.prank(_actor);
+    vm.expectEmit();
+    emit ProposalCreated(proposalId, _actor, targets, values, signatures, calldatas, voteStart, voteEnd, description);
+    governor.propose(targets, values, calldatas, description);
+
+    assertEq(uint256(governor.state(proposalId)), uint256(IGovernor.ProposalState.Pending));
+  }
+}
```
Comment (wildmolasses) [inline test/L2ArbitrumGovernorV2.t.sol:241]: shall we add other proposal tests? e.g. RevertIf_DoesNotMeetProposalThreshold? I think this would be a different PR if we chose to do it.

File: test/L2ArbitrumGovernorV2.t.sol [hunk 3]
```diff
@@ -196,6 +206,96 @@ abstract contract Quorum is L2ArbitrumGovernorV2Test {
...
+
+    vm.prank(_proposer);
+    governor.propose(targets, values, calldatas, description);
+
+    vm.prank(address(_actor));
+    vm.expectRevert(abi.encodeWithSelector(GovernorOnlyProposer.selector, _actor));
+    governor.cancel(targets, values, calldatas, keccak256(bytes(description)));
+  }
+
+  function testFuzz_RevertIf_ProposalIsActive(uint256 _actorSeed) public {
```
Comment (wildmolasses) [inline test/L2ArbitrumGovernorV2.t.sol:281]: wonder if we can use `mockCall` to simulate all proposal states besides `Pending`

File: test/L2ArbitrumGovernorV2.t.sol [hunk 4]
```diff
@@ -196,6 +206,96 @@ abstract contract Quorum is L2ArbitrumGovernorV2Test {
...
+  }
+}
+
+abstract contract Cancel is L2ArbitrumGovernorV2Test {
+  event ProposalCanceled(uint256 proposalId);
+
+  error GovernorOnlyProposer(address proposer);
+  error ProposalNotPending(IGovernor.ProposalState state);
+
+  function testFuzz_CancelProposalWhenPending(uint256 _actorSeed) public virtual {
```
Comment (wildmolasses) [inline test/L2ArbitrumGovernorV2.t.sol]:
```suggestion
  function testFuzz_CancelsPendingProposal(uint256 _actorSeed) public virtual {
```

**Review by wildmolasses — COMMENTED** [review]
thanks @garyghayrat a few nits

File: test/L2ArbitrumGovernorV2.t.sol (hunk 2 above)
Comment (garyghayrat) [reply test/L2ArbitrumGovernorV2.t.sol:241]: Yes! Will do it in another PR.

File: test/L2ArbitrumGovernorV2.t.sol (hunk 3 above)
Comment (garyghayrat) [reply test/L2ArbitrumGovernorV2.t.sol:281]: oh good idea!

**Review by wildmolasses — APPROVED** [review]
lgtm

File: test/L2ArbitrumGovernorV2.t.sol (hunk 4 above)
Comment (garyghayrat) [reply test/L2ArbitrumGovernorV2.t.sol]: https://github.com/ScopeLift/arbitrum-governor-upgrade/pull/9/commits/9d2cfc7e690791e27de0780533d5062a840f9970

File: test/L2ArbitrumGovernorV2.t.sol (hunk 1 above)
Comment (garyghayrat) [reply test/L2ArbitrumGovernorV2.t.sol]: https://github.com/ScopeLift/arbitrum-governor-upgrade/pull/9/commits/9d2cfc7e690791e27de0780533d5062a840f9970
